pipeline/build.go

```diff
@@ -33,34 +33,54 @@ type Build struct {
 // function will remove the steps that have a ruleset that do not
 // match the provided ruledata. If both stages and steps are
 // provided, then an empty pipeline is returned.
-func (b *Build) Purge(r *RuleData) *Build {
+func (b *Build) Purge(r *RuleData) (*Build, error) {
```

Comment:
from first glance i didnt see a PR to update the method call in server
https://github.com/go-vela/server/blob/main/compiler/native/expand.go#L92

i wonder if returning `err` has an underlying complication with that `continue`? because we wouldve normally just returned an empty pipeline and moved on but now we have a couple different errors that might occur

Reply:
im not sure though, ill have to test that myself with some bogus pipelines

Reply:
tested it out, it bubbles up to an audit failure very nicely. nice work